Zend/zend_compile.c

```diff
@@ -11232,6 +11238,52 @@ static void zend_compile_const_expr_closure(zend_ast **ast_ptr)
 	*ast_ptr = zend_ast_create_op_array(op);
 }
 
+static void zend_compile_const_expr_fcc(zend_ast **ast_ptr)
+{
+	zend_ast **args_ast;
+	switch ((*ast_ptr)->kind) {
+		case ZEND_AST_CALL:
+			args_ast = &(*ast_ptr)->child[1];
+			break;
+		case ZEND_AST_STATIC_CALL:
+			args_ast = &(*ast_ptr)->child[2];
+			break;
+		EMPTY_SWITCH_DEFAULT_CASE();
+	}
+	if ((*args_ast)->kind != ZEND_AST_CALLABLE_CONVERT) {
+		zend_error_noreturn(E_COMPILE_ERROR, "Constant expression contains invalid operations");
+	}
+	ZEND_MAP_PTR_NEW(((zend_ast_fcc *)*args_ast)->fptr);
+
+	switch ((*ast_ptr)->kind) {
+		case ZEND_AST_CALL: {
+			zend_ast *name_ast = (*ast_ptr)->child[0];
+			zval *name_ast_zv;
+			if (name_ast->kind != ZEND_AST_ZVAL || Z_TYPE_P(name_ast_zv = zend_ast_get_zval(name_ast)) != IS_STRING) {
```

Comment:
Nit: Extract `name_ast_zv = zend_ast_get_zval(name_ast)` for easier visibility?

Reply:
Good catch: The non-string case should also result in a different error message (“Invalid function name” for consistency with `zend_resolve_class_name_ast()`). I had initially written it like this so that I do not need to duplicate the error message.

Will be fixed + test added.